### ISSUE TITLE
Disallow event handler shorthand syntax

### ISSUE THREAD

**Issue by edg2s** [wikimedia/eslint-config-wikimedia#127]
Aside from saving a few bytes, using the event handler shorthand syntax seems like a bad idea:
* #on: `$el.on( 'keydown', handler );`
* shorthand: `$el.keydown( handler );`

The advantages of the on syntax:
* Makes it clear that event names are just string properties being pass around
* Aligns with `.off ( 'keydown', handler )` syntax
* Aligns with `.addEventListener( 'keydown', handler )` native syntax

Disadvantages:
* A few bytes longer

The plugin rule `jquery/no-event-shorthand` disables all these methods. It also disables using them as triggers, preferring `$el.trigger( 'focus' )` over `$el.focus()`.

This case seems less egregious, as there the on/off argument doesn't apply, but probably isn't worth separating.

Also because the way the linter works, `$el[ 0 ].focus()` will get picked up as a false positive, but that can be worked around (`el = $el[ 0 ]; el.focus()`).

**Comment by MatmaRex** [comment]
> It also disables using them as triggers, preferring $el.trigger( 'focus' ) over $el.focus().

Pretty sure these are not equivalent. The first will only run jQuery's event handlers. The second will also call the native .focus() method. (Similarly for .click() and maybe others.)

**Comment by edg2s** [comment]
> > It also disables using them as triggers, preferring $el.trigger( 'focus' ) over $el.focus().
> 
> Pretty sure these are not equivalent. The first will only run jQuery's event handlers. The second will also call the native .focus() method. (Similarly for .click() and maybe others.)

Looking at the code they seem to be identical:
```javascript
jQuery.each( ( "blur focus focusin focusout resize scroll click dblclick " +
	"mousedown mouseup mousemove mouseover mouseout mouseenter mouseleave " +
	"change select submit keydown keypress keyup contextmenu" ).split( " " ),
	function( i, name ) {

	// Handle event binding
	jQuery.fn[ name ] = function( data, fn ) {
		return arguments.length > 0 ?
			this.on( name, null, data, fn ) :
			this.trigger( name );
	};
} );
```

When arguments.length is 0, it just calls this.trigger( name );

**Comment by MatmaRex** [comment]
Looks like I was confusing .trigger() and .triggerHandler(). Never mind me.
